lib/_http_client.js

```diff
@@ -328,7 +325,7 @@ function socketOnEnd() {
   if (!req.res && !req.socket._hadError) {
     // If we don't have a response then we know that the socket
     // ended prematurely and we need to emit an error on the request.
-    req.emit('error', createHangUpError());
+    process.nextTick(() => req.emit('error', createHangUpError()));
```

Comment:
This has the side effect of changing the stack where the hangup error is created (the `new Error` is called inside the callback on next tick). This will disassociate the stack trace from it's context. If we're going to do this, I recommend first creating the Error object, *then* passing it off to the next tick for dispatch and handling...

```
const err = createHangUpError();
process.nextTick(() => req.emit('error', err));
```